crates/prices/src/feed.rs

```diff
@@ -0,0 +1,55 @@
+use serde::{
+    de::{self, Visitor},
+    Deserialize,
+};
+
+use super::error::Error;
+use super::chainlink::ChainlinkFeed;
+
+struct FeedVisitor {}
+
+impl<'de> Visitor<'de> for FeedVisitor {
+    type Value = Feed;
+
+    fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+        formatter.write_str("Could not deserialize feed")
+    }
+
+    fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+    where
+        A: serde::de::SeqAccess<'de>,
+    {
+        let provider = seq.next_element()?.unwrap();
+        let address = seq.next_element()?.unwrap();
+        match Feed::create(provider, address) {
+            Ok(value) => Ok(value),
+            Err(other) => return Err(de::Error::custom(other)),
+        }
+    }
+}
```

Comment:
I'm pretty sure you didn't have to implement your own deserializer 😅 usually you should be able to go with just `#[derive(Deserialize)]`. if you can't, it either means you're doing some more complex parsing (not the case here), or you're not setting the format properly

Reply:
my objective was to automatically deserialize into one of the `Feed` enum depending on the first string of the vector.  I'll take a better look on how to do this then.